tasks/libs/releasing/version.py

```diff
@@ -370,14 +370,30 @@ def get_version_numeric_only(ctx, major_version='7'):
     return version
 
 
-def load_dependencies(_):
-    with open("release.json") as f:
-        versions = json.load(f)
-        if RELEASE_JSON_DEPENDENCIES not in versions:
-            raise Exception(f"Could not find '{RELEASE_JSON_DEPENDENCIES}' in release.json")
-        # windows runners don't accepts anything else than strings in the
-        # environment when running a subprocess.
-        return {str(k): str(v) for k, v in versions[RELEASE_JSON_DEPENDENCIES].items()}
+def load_overridden_dependencies():
+    """
+    Load dependency versions from release.json with environment variable overrides.
+    WINDOWS_* dependencies are skipped on non-Windows platforms.
+
+    Returns:
+        dict: Environment dictionary with dependency versions as strings.
+    """
+    from tasks.libs.releasing.json import load_release_json  # delayed to avoid circular imports
+
+    release = load_release_json()
+    if not (dependencies := release.get(RELEASE_JSON_DEPENDENCIES)):
+        raise Exception(f"Could not find {RELEASE_JSON_DEPENDENCIES!r} in release.json")
+    env = {}
```

Comment:
Nit: I'd suggest more clear/consistent naming here, such as `final_dependencies` or some such (or renaming `dependencies` to`static_dependencies` or something like that). Alternatively the name of the function could be updated to reflect that this returns an "env".